apps/api/src/domain/games/memory-match/services/memory-match.service.ts

```diff
@@ -157,4 +241,99 @@ export class MemoryMatchService {
             'NODE_PORT'
         )}/${this.configService.get('GLOBAL_PREFIX')}/resources/mediaItems/download/${mediaItemId}`;
     }
+
+    private async validateMediaItemReferences(
+        memoryMatchRound: MemoryMatchRound
+    ): Promise<InternalError[]> {
+        if (!isNonEmptyObject(memoryMatchRound)) {
+            return [];
+        }
+
+        const { cardBackImageId } = memoryMatchRound;
+
+        const imageIds = memoryMatchRound.cards.map(({ imageId }) => imageId);
+
+        if (!isNullOrUndefined(cardBackImageId)) {
+            imageIds.push(cardBackImageId);
+        }
+
+        const audioIds = memoryMatchRound.cards.map(({ audioId }) => audioId);
+
+        const mediaItemIds = [...imageIds, ...audioIds];
```

Comment:
I wonder if we want to have a helper method on the `MemoryMatchRound` to gather all media item references?